src/main/java/properties/ReadProperties.java

```diff
@@ -1,30 +1,44 @@
 
 package properties;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import java.io.*;
 import java.util.Properties;
 
-public class ReadProperties {
 
+public final class ReadProperties {
+
+    private static final Logger logger = LogManager.getLogger();
     private final Properties properties = new Properties();
+    private static ReadProperties instance;
+
+    public static ReadProperties getInstance() {
+        if (instance == null) {
+            instance = new ReadProperties();
+        }
+        return instance;
+    }
 
-    public ReadProperties() {
+    private ReadProperties() {
         try (InputStream inputStream = getClass().getClassLoader().getResourceAsStream("config.properties")) {
             properties.load(inputStream);
         } catch (IOException e) {
+            logger.debug("The file cannot be read...");
             e.printStackTrace();
         }
     }
 
     /***
      * Receives the key value from config.properties
-     * @param key
+     * @param key key from config.properties
      * @return key value
      */
-    public String getPropertyValueByKey(String key) {
+    public String getKeyProperty(String key) {
```

Comment:
I think it makes sense to create a enum for PropertyKey and pass it as parameter here instead of String in order to exclude issues with wrong parameters keys

Reply:
done